Zend/zend_closures.c

```diff
@@ -515,16 +515,32 @@ static HashTable *zend_closure_get_debug_info(zend_object *object, int *is_temp)
 
 	if (closure->func.type == ZEND_USER_FUNCTION && closure->func.op_array.static_variables) {
 		zval *var;
+		zend_string *key;
 		HashTable *static_variables =
-			ZEND_MAP_PTR_GET(closure->func.op_array.static_variables_ptr);
-		ZVAL_ARR(&val, zend_array_dup(static_variables));
-		zend_hash_update(debug_info, ZSTR_KNOWN(ZEND_STR_STATIC), &val);
-		ZEND_HASH_FOREACH_VAL(Z_ARRVAL(val), var) {
-			if (Z_TYPE_P(var) == IS_CONSTANT_AST) {
-				zval_ptr_dtor(var);
-				ZVAL_STRING(var, "<constant ast>");
+			zend_array_dup(ZEND_MAP_PTR_GET(closure->func.op_array.static_variables_ptr));
```

Comment:
No point in dup'ing them anymore.

Reply:
I thought that and tried that first, and got strange results, I'll look again ...

Reply:
Ah, I didn't handle the IS_REF && REFCOUNT == 1